components/blitz/src/omero/cmd/fs/FindPyramidsI.java

```diff
@@ -0,0 +1,226 @@
+        sb.append("select p from Pixels as p ");
+        sb.append("left outer join fetch p.image as i ");
+        sb.append("left outer join fetch i.details.creationEvent ");
+        sb.append("where p.id = :id");
+        query = sb.toString();
+        walkDirectory(pixeldsDir);
+        rsp.pyramidFiles = imageIds;
+    }
+
+    /**
+     * Walks the specified directory.
+     *
+     * @param dir The directory to walk.
+     */
+    private void walkDirectory(File dir) {
+        File[] files = dir.listFiles();
+        if (files == null) {
+            return;
+        }
+        for (File file : files) {
+            if (file.isDirectory()) {
+                walkDirectory(file);
+            } else {
+                String name = file.getName();
+                if (name.endsWith("_pyramid")) {
+                    boolean toDelete = true;
+                    if (checkEmptyFile) {
+                        if (file.length() == 0) {
+                            File[] list = pixeldsDir.listFiles();
+                            for (File lockfile : list) {
+                                String n = lockfile.getName();
+                                if (n.startsWith("." + name) &&
+                                (n.endsWith(".tmp") || n.endsWith(".pyr_lock"))) {
+                                    toDelete = false;
+                                    break;
+                                }
+                                
+                            }
+                        }
+                    }
+                    if (toDelete) {
+                        String[] values = name.split("_");
+                        long id = getImage(Long.parseLong(values[0]));
+                        if (id > 0) {
+                            imageIds.add(id);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Finds the image matching the criteria.
+     * @param pixelsId The pixels set ID.
+     * @return The image Id or -1.
+     */
+    private long getImage(long pixelsId) {
+        Pixels pixels = service.findByQuery(query, new Parameters().addId(pixelsId));
+        Image image = pixels.getImage();
+        if (checkEmptyFile) {
+            return image.getId().longValue();
+        }
+        long creation = image.getDetails().getCreationEvent().getTime().getTime();
+        if (importeAfter < creation/1000) {
+            if (isLittleEndian(pixels) == littleEndian) {
+                return image.getId().longValue();
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Returns whether or not the pixels set is little endian.
+     * @param pixels The pixels set to handle
+     * @return See above.
+     */
+    private boolean isLittleEndian(Pixels pixels) {
+        try {
+            PixelBuffer pf = pixelsService._getPixelBuffer(pixels, false);
+            if (pf instanceof BfPixelsWrapper) {
```

Comment:
This block still concerns me. Perhaps with the work on the OmeroPyramidWriter code, we can simplify it.

Reply:
I don't like it either but unfortunately I could not introduce a ``isLittleEndian`` method (b/c to the way RomioPixelBuffer works)
I will add a TODO so we can go back to it